extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java

```diff
@@ -242,15 +242,55 @@ private String getRequestAccept(RoutingContext ctx) {
         return DEFAULT_RESPONSE_CONTENT_TYPE;
     }
 
-    private String getCharset(String mimeType) {
-        if (mimeType != null && mimeType.contains(";")) {
-            String[] parts = mimeType.split(";");
-            for (String part : parts) {
-                if (part.trim().startsWith("charset")) {
-                    return part.split("=")[1];
-                }
+    private static int indexOfSemicolonOrWhitespace(String s, int start) {
+        for (int i = start; i < s.length(); i++) {
+            final char c = s.charAt(i);
+            if (c == ';' || Character.isWhitespace(c)) {
+                return i;
             }
         }
+        return -1;
+    }
+
+    private static String parseCharset(String mimeType) {
+        if (mimeType == null) {
+            return null;
+        }
+        final int charsetIndex = mimeType.indexOf("charset=");
+        if (charsetIndex == -1) {
+            return null;
+        }
+        final int charsetValueStart = charsetIndex + 8;
+        final int firstSemicolonWhitespace = indexOfSemicolonOrWhitespace(mimeType, charsetValueStart);
+        final int charsetValueEnd = firstSemicolonWhitespace == -1 ? mimeType.length() : firstSemicolonWhitespace;
+        final int charsetLen = charsetValueEnd - charsetValueStart;
+        if (charsetLen == 0) {
+            return null;
+        }
+        return charsetOf(mimeType, charsetValueStart, charsetLen);
+    }
+
+    private static String charsetOf(String mimeType, int charsetValueStart, int charsetLen) {
```

Comment:
@geoand we could add others too eg upper-case versions of the same charset(s)
I'm not aware which "common" charset names are setup on the wire for this. Maybe GraphQL folks knows?

Reply:
So there are not typical values used other than utf-8 eg us-ascii or uppercased versions?

Reply:
Right, at least I'm not aware of anything

Reply:
I've updated the code to cache just the UTF-8/utf-8 cases

eg
```
application/json;q=0.9, text/plain;q=0.5; charset=utf-8


application/json;q=0.9, text/plain;q=0.5; charset=UF-8
```